src/r_things.c

```diff
@@ -1053,15 +1086,82 @@ void R_DrawMasked(void)
 {
   int i;
   drawseg_t *ds;
+  const int cx = MAX_SCREENWIDTH / 2;
 
   R_SortVisSprites();
 
+  // [Woof!] e6y
+  // Reducing of cache misses in the following R_DrawSprite()
+  // Makes sense for scenes with huge amount of drawsegs.
+  // ~12% of speed improvement on epic.wad map05
+  for(i = 0; i < DS_RANGES_COUNT; i++)
+    drawsegs_xranges[i].count = 0;
+
+  if (num_vissprite > 0)
+  {
+    if (drawsegs_xrange_size < maxdrawsegs)
+    {
+      drawsegs_xrange_size = 2 * maxdrawsegs;
+      for(i = 0; i < DS_RANGES_COUNT; i++)
+      {
+        drawsegs_xranges[i].items = realloc(
```

Comment:
We removed macros for memory management functions, so we have to use `Z_Realloc` here (this is `Z_Realloc` in PrBoom+ too, just hidden behind the macro). Not sure about that, thought @fabiangrefrath what do you think?

Reply:
Yes, let's explicitly use Z_Realloc here.